sql: don't overwrite ctx with nil on failure

asubiotto opened the pull request:
SetupLocalSyncFlow would return a nil ctx with an error in setupFlows if an
error occurred. The previous version would overwrite a ctx in the outer scope
just to return it. This commit no longer overwrites ctx and simply returns the
new ctx when there is no error. On master, this nil ctx would not have any
consequences although this commit is good hygiene, but 20.2 uses this ctx in a
defer, which would cause a panic.

Release note: None (no panic present on master)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/59002)
<!-- Reviewable:end -->

yuzefovich reviewed (commented):
Hm, I guess that what I tried to fix with #58315.

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59002#-:-MR11D8L-A3pdhTRMPI5:befciw5)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @asubiotto)

___
*[pkg/sql/distsql_running.go, line 227 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/59002#-MR11IMD0VReD3WBKlO2:-MR11IMD0VReD3WBKlO3:b-44mziz) ([raw file](https://github.com/cockroachdb/cockroach/blob/c6c50cf45411f14802ac94c2f0528a46df63b4df/pkg/sql/distsql_running.go#L227)):*
> ```Go
> 	localReq.Flow = *flows[thisNodeID]
> 	defer physicalplan.ReleaseSetupFlowRequest(&localReq)
> 	newCtx, flow, err := dsp.distSQLSrv.SetupLocalSyncFlow(ctx, evalCtx.Mon, &localReq, recv, localState)
> ```

I wonder why would we not just do `return dsp.distSQLSrv.SetupLocalSyncFlow(ctx, evalCtx.Mon, &localReq, recv, localState)`? The single caller is checking the error right away too.


<!-- Sent from Reviewable.io -->

Commit:
sql: don't overwrite ctx with nil on failure

SetupLocalSyncFlow would return a nil ctx with an error in setupFlows if an
error occurred. The previous version would overwrite a ctx in the outer scope
just to return it. This commit no longer overwrites ctx and simply returns the
new ctx. On master, this nil ctx would not have any consequences although this
commit is good hygiene, but 20.2 uses this ctx in a defer, which would cause a
panic.

Release note (bug fix): Fix a nil pointer panic edge case in query setup code.

asubiotto commented:
Good point. Done.

yuzefovich reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

bors r=yuzefovich

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59002#-:-MR5kDlt3xL3uTofDncX:b-mwcsh8)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @asubiotto)



<!-- Sent from Reviewable.io -->
